src/docserv/functions.py

```diff
@@ -32,6 +33,49 @@ def mail(text, subject, to):
     p.communicate(msg.as_string())
 
 
+def get_all_file_paths(directory, zip_formats):
+    """Collect all files with paths in specified formats from given directory.
+
+    :param str directory: a directory from which files are to be collected
+    :param list zip_formats: a list containing accepted documentation formats
+        example: ["pdf", "epub", "single-html"]
+    :return: a list with all files matching specified documentation formats
+    :rtype: list
+    """
+    file_paths = []
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            format_ok = 0
+            for format in zip_formats:
+                if format in filepath:
+                    format_ok = 1
+                elif root is directory and format in filepath:
+                    root_ok = 1
+            if format_ok:
+                file_paths.append(filepath)
+    return file_paths
```

Comment:
The basic algorithm looks good, but I have only one improvement. :wink: 

For things like returning a list of files etc. it's usually a good idea to use a *generator*. So you don't return the complete list, but the the current item at the time. The changes are minimal:

```python
for root, directories, files in os.walk(directory):
    for filename in files:
        filepath = os.path.join(root, filename)
        format_ok = 0
        for format in zip_formats:
            if format in filepath:
                format_ok = 1
        if format_ok:
            yield filepath
```

Reply:
Thanks! Changes made